domain/schema/model/sql/0020-network.sql

```diff
@@ -23,3 +23,163 @@ CREATE TABLE cloud_container (
     FOREIGN KEY (net_node_uuid)
     REFERENCES net_node (uuid)
 );
+
+CREATE TABLE cloud_container_ip_address (
+    net_node_uuid TEXT NOT NULL,
+    ip_address_uuid TEXT NOT NULL,
+    CONSTRAINT fk_cloud_container_ip_address_net_node_uuid
+    FOREIGN KEY (net_node_uuid)
+    REFERENCES cloud_container (net_node_uuid),
+    CONSTRAINT fk_cloud_container_ip_address_ip_address_uuid
+    FOREIGN KEY (ip_address_uuid)
+    REFERENCES ip_address (uuid),
+    PRIMARY KEY (net_node_uuid, ip_address_uuid)
+);
+
+CREATE TABLE cloud_container_port (
+    net_node_uuid TEXT NOT NULL,
+    port TEXT NOT NULL,
+    CONSTRAINT fk_cloud_container_port_net_node_uuid
+    FOREIGN KEY (net_node_uuid)
+    REFERENCES cloud_container (net_node_uuid),
+    PRIMARY KEY (net_node_uuid, port)
+);
+
+-- ip_address_type represents the possible ways of specifying
+-- an address, either a hostname resolvable by dns lookup,
+-- or IPv4 or IPv6 address.
+CREATE TABLE ip_address_type (
+    id INT PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_ip_address_type_name
+ON ip_address_type (name);
+
+INSERT INTO ip_address_type VALUES
+(0, 'hostname'),
+(1, 'ipv4'),
+(2, 'ipv6');
+
+-- ip_address_origin represents the authoritative source of
+-- an ip address.
+CREATE TABLE ip_address_origin (
+    id INT PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_ip_address_origin_name
+ON ip_address_origin (name);
+
+INSERT INTO ip_address_origin VALUES
+(0, 'unknown'),
+(1, 'machine'),
+(2, 'provider');
+
+-- ip_address_scope denotes the context an address may apply to.
+-- If a name or address can be reached from the wider internet,
+-- it is considered public. A private network address is either
+-- specific to the cloud or cloud subnet a machine belongs to,
+-- or to the machine itself for containers.
+CREATE TABLE ip_address_scope (
+    id INT PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_ip_address_scope_name
+ON ip_address_scope (name);
+
+INSERT INTO ip_address_scope VALUES
+(0, 'unknown'),
+(1, 'public'),
+(2, 'local-cloud'),
+(3, 'local-machine'),
+(4, 'link-local');
+
+-- ip_address_config_type defines valid network
+-- link configuration types.
+CREATE TABLE ip_address_config_type (
+    id INT PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_ip_address_config_type_name
+ON ip_address_config_type (name);
+
+INSERT INTO ip_address_config_type VALUES
+(0, 'unknown'),
+(1, 'dhcp'),
+(2, 'static'),
+(3, 'manual'),
+(4, 'loopback');
+
+CREATE TABLE ip_address (
+    uuid TEXT NOT NULL PRIMARY KEY,
+    -- The value of the configured IP address.
+    -- e.g. 192.168.1.2 or 2001:db8::/64.
```

Comment:
nit: This comment is confusing. You have supplied the v4 address here without a CIDR and the v6 address with a CIDR. Does this address carry a CIDR or not ?

Reply:
Comment copied from mongo, I'll tidy it up
```
	// Value is the value of the configured IP address, e.g. 192.168.1.2 or
	// 2001:db8::/64.
	Value string `bson:"value"`
```